src/controllers/dynakube/oneagent/daemonset/arguments.go

```diff
@@ -15,12 +15,27 @@ const (
 )
 
 func (dsInfo *builderInfo) arguments() []string {
-	metadata := deploymentmetadata.NewDeploymentMetadata(dsInfo.clusterId, dsInfo.deploymentType)
-	args := dsInfo.hostInjectSpec.Args
+
+	args := make([]string, 0)
+
+	args = dsInfo.appendHostInjectArgs(args)
 	args = dsInfo.appendProxyArg(args)
 	args = dsInfo.appendNetworkZoneArg(args)
 	args = appendOperatorVersionArg(args)
-	args = append(args, metadata.AsArgs()...)
+	args = dsInfo.appendMetadataArgs(args)
```

Comment:
Would have moved this line up by one, s.t. all ds metadata are appended before something else, but guess this is just preference

Reply:
i will leave it as is since it is a cherry pick, otherwise, yes, it would've indeed looked nicer